user/views.py

```diff
@@ -3,11 +3,13 @@
 from django.utils import timezone
 from .forms import ShowForm
 from django.views import View
+from django.contrib.auth.decorators import login_required
 
 def shows(request):
     upcoming_shows = Show.objects.all().order_by('-time')
     return render(request, 'shows.html', {'shows': upcoming_shows})
 
+@login_required
```

Comment:
If I implement a login for users as well wont this mean all authenticated users can add movies? I want only the superuser or staff https://docs.djangoproject.com/en/5.1/topics/auth/default/#user-objects to be able to add movies. Read more about the implementation here https://stackoverflow.com/questions/12003736/django-login-required-decorator-for-a-superuser

Reply:
I looked into it, doesn't looks like the login_required will check for superuser. So how about a custom decorator maybe?.

Reply:
Ya thats sounds good

Reply:
Also do a git pull before making the next commit.

Reply:
Okay! so turns out there is a staff_member_required decorator baked inside django. It can restrict regular users to access add_movie function based on their staff member status, redirecting them to the original admin login panel. Even if we create a user with User model within a view it has all the admin access set to false by default & inverse is true for a superuser created from terminal. Its either has access or no access at all. Does its implementation with staff_member_required sound fine?